### DISCUSSION TITLE
Is Modal focus trap too restrictive?

### DISCUSSION THREAD

**Discussion by lucsky** [adobe/react-spectrum#6372] [Q&A] [answered]
Greetings, my company has an accessibility rule which seems to be in direct conflict with the default behaviour of RAC `Modal` focus trap. The rule basically states that a user must always be able to move focus out of a subset of content, **and be able to access the browser toolbars** (emphasis mine). It obviously make sense to isolate children of modal containers from outside element *of the page itself*, but what about the browser's elements? 

Is there any way around that?

Thanks!

**Comment by devongovett** [comment, marked as answer]
There is some debate about this within the accessibility community. For example, see this thread on ARIA Practices: https://github.com/w3c/aria-practices/issues/1772. I think the general consensus there is that allowing tab to cycle out of the modal into the browser chrome is confusing and potentially frustrating for users because they need to tab through every button and other element in the toolbar (which could be a lot, e.g. bookmarks) to get back to the start of the modal. In addition, allowing users to tab to the browser chrome, but not tab out of iframes to other content on the page is pretty much impossible to implement. We generally think modals should be modal no matter their context, and for those reasons, we do not include the browser chrome in the tab sequence. Users have other ways of reaching the browser chrome such as Cmd/Ctrl + L to jump to the address bar and tabbing from there, and other keyboard shortcuts can be used to navigate forward and back, etc.

**Reply by lucsky** [reply]
Yeah, I think I tend to agree with the consensus, thanks for the pointers!
